### ISSUE TITLE
BUG: Xorbits.numpy raises TypeError

### ISSUE THREAD

**Issue by WuTianyi321** [xprobe-inc/xorbits#318]
Reproduce:
```import xorbits
import xorbits
xorbits.init()
from xorbits import numpy as np
a=np.ones(5,dtype=bool)
b=np.zeros(5,dtype=bool)
if np.any(a&b):
    print(1)
```

Full stack:
```
IndexError                                Traceback (most recent call last)
File ~/Documents/projects/xorbits/python/xorbits/core/data.py:221, in DataRef.__len__(self)
    220 try:
--> 221     return int(self.data._mars_entity.shape[0])
    222 except (IndexError, ValueError):
    223     # shape is unknown, execute it

IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
File ~/Documents/projects/xorbits/python/xorbits/core/data.py:226, in DataRef.__len__(self)
    225 try:
--> 226     return int(self.data._mars_entity.shape[0])
    227 except (IndexError, ValueError):  # happens when dimension is 0

IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
Cell In[2], line 5
      3 a=np.ones(5,dtype=bool)
      4 b=np.zeros(5,dtype=bool)
----> 5 if np.any(a&b):
      6     print(1)

File ~/Documents/projects/xorbits/python/xorbits/core/data.py:228, in DataRef.__len__(self)
    226             return int(self.data._mars_entity.shape[0])
    227         except (IndexError, ValueError):  # happens when dimension is 0
--> 228             raise TypeError(
    229                 f"object with shape {self.data._mars_entity.shape} has no len()"
    230             )
    231 else:
    232     raise TypeError(f"object of type '{self.data.data_type}' has no len()")

TypeError: object with shape () has no len()
```

OS: Windows 10 
python: Python 3.8.8
xorbits: 0.2.1

**Comment by qianduoduo0904** [comment]
Could you please translate the title and content into English, so that everyone from all over world can understand. Thank you in advance.

**Comment by Zhou1213CN** [comment]
Take

**Comment by qianduoduo0904** [comment]
This issue has been addressed on main branch, and will be included in v0.3.0 release.
